tasks/install-matlab/v1-beta/src/matlab.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023 The MathWorks, Inc.
+
+import * as toolLib from "azure-pipelines-tool-lib/tool";
+import * as fs from "fs";
+import * as https from "https";
+import * as script from "./script";
+
+export interface Release {
+    name: string;
+    version: string;
+    update: string;
+}
+
+export async function makeToolcacheDir(release: Release): Promise<[string, boolean]> {
+    let toolpath: string = toolLib.findLocalTool("MATLAB", release.version);
+    let alreadyExists = false;
+    if (toolpath) {
+        alreadyExists = true;
+    } else {
+        fs.writeFileSync(".keep", "");
+        toolpath = await toolLib.cacheFile(".keep", ".keep", "MATLAB", release.version);
+    }
+    return [toolpath, alreadyExists];
+}
+
+export async function getReleaseInfo(release: string): Promise<Release> {
+    // Get release name from input parameter
+    let name: string;
+    if ( release.toLowerCase().trim() === "latest") {
+        try {
+            name = await resolveLatest();
+        } catch {
+            return Promise.reject(Error(`Unable to retrieve the MATLAB release information. Contact MathWorks at continuous-integration@mathworks.com if the problem persists.`));
+        }
+    } else {
+        const nameMatch = release.toLowerCase().match(/r[0-9]{4}[a-b]/);
+        if ( !nameMatch ) {
+            return Promise.reject(Error(`${release} is invalid or unsupported. Specify the value as R2020a or a later release.`));
+        }
+        name = nameMatch[0];
+    }
+
+    // create semantic version of format year.semiannual.update from release
+    const year = name.slice(1, 5);
+    const semiannual = name[5] === "a" ? "1" : "2";
+    const updateMatch = release.toLowerCase().match(/u[0-9]/);

```

Comment:
```suggestion
    const updateMatch = release.toLowerCase().match(/u[0-9]+/);
```
A bit safer since we're already nearing double digit updates.

Reply:
Good catch